config/federationConfigClass.js

```diff
@@ -0,0 +1,130 @@
+ * @access reserved
+ * @async fsct
+ * @author @wojtekLs
+ * @class federationConfigClass
+ * @copyright @wojtekLs
+ * @version 0.1
+ * @see https://github.com/wojtekLs/ejs/pull/3
+ */
+export default class federationConfigClass {
+	super(
+		id = 0,
+		imageId = "",
+		imageType = Object,
+		deploymentZone = Object,
+		tmplTypeName = "",
+		serverName = Object,
+		metaData = Object
+	);
+
+	/**
+	 * @version 0.1-alpha
+	 * @todo #4 Add logic for programatically creating service worker (as a file)
+	 */
+	openApiEndServiceWorkerCA() {
+		const SwRegistration = async () => {
+			if (worker in navigator) {
+				try {
+					const reg = await navigator.serviceWorker.register(worker);
+				} catch (e) {
+					winston.error(workerRegistrationError);
+				}
+			} else {
+				winston.error(noServiceWorkerExist);
+			}
+		}
+	}
+
+	/**
+	 * FAS = Federation Administrative Server 
+	 */
+	getFAServerInstanceData() {
+		return {
+			id: [0, 255],
+			imageId: String,
+			imageType: localStorage.getItem('img-type-string'),
+			deploymentZone: localStorage.getItem('instance-deployment-zone-intl'),
+			tmplTypeName: String,
+			serverName: localStorage.getItem('srv-name'),
+			metaData: [String]
+		}
+	}
+
+	/**
+	 * Generate random ID for API to use
+	 * @author @wojtekLs
+	 * @class federationConfigClass
+	 * @version 0.1
+	 */
+	generateID(max) {
+		return this.super.id = Math.floor(Math.random() * max);
+	}
+
+	/**
+	 * @todo: use try->catch.
+	 * 
+	 * @author @wojtekLs
+	 * @version 0.1
+	 */
+	setFAServerInstanceData() {
+		this.super.id = this.generateID(6);
+		this.super.imageId = document.getElementById('imgIdBox-slector').ariaSelected;
+		this.super.imageType = localStorage.setItem('img-type-string', document.getElementById('imgTypeInputBox').ariaSelected);
+		this.super.deploymentZone = localStorage.setItem('instance-deployment-zone-intl', document.getElementById('instanceDplyZone').ariaSelected);
+		this.super.tmplTypeName = window.FS.readFile(paths.defaultTemplateFile, "utf-8", (err, data) => {
+			if (err) {
+				winston.error(errorReadingTmplFile);
+				return;
+			}
+		});
+		this.super.serverName = localStorage.setItem('srv-name', document.getElementById('serverNameText').ariaSelected);
+		// @todo: #5 add metadata proccessing code
+	}
+
+	/**
+	 * Writes schema to JSON file
+	 * 
+	 * @param {} federationSchemaName
+	 * @param {*} federationSchemaFile
+	 * @todo #6 Upload schema file to SchemaStore
+	 * @author @wojtekLs
+	 * @version 0.1
+	 */
+	writeSchemaJSONFile(federationSchemaName, federationSchemaFile) {
+		return {
+			url: "https://www.schemastore.org/json/" + federationSchemaName,
+			data: { url },
+			success: (data) => {
+				let binary = Buffer.from(data, "base64");
+				let array = [];
+				for (let i = 0; i < binary.length; i++) {
+					array.push(binary[i]);
+				}
+				federationSchemaFile = new Blob([new Uint8Array(array)], { type: 'application/json' });
+			},
+			fail: () => {
+				winston.error(errorSettingSchema);
```

Comment:
```suggestion
				winston.error(errorSettingSchema);
```
```suggestion
				winston.error(errorSettingSchema);
```edit it to log to `stdout` instead